feat: Page or Modal to change the password

DanielEJaimes opened the pull request:
Modal with the tailwind implemetation
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/b8a4c8dc-49ff-486f-8674-4c44b3d1c5ac)
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/72e85504-8efb-4f57-9dcb-fee84d625e5d)

DanielEJaimes committed:
feat: Page or Modal to change the password

Added the library MUI
In the main file the commented section is for test

PedroChaparro reviewed (commented):
My only comment is that you're using a huge library (material) just for a modal and the style of the inputs aren´t consistent with the ones done by @MiguelMRojas in #20. 

Anyways, you're free to use any library. @MiguelMRojas please review this PR.

MiguelMRojas reviewed (changes requested):
I agree with @PedroChaparro, I also feel that using a library is overkill, when a modal can be done with tailwind directly,  [https://www.creative-tim.com/learning-lab/tailwind-starter-kit/documentation/react/modals/regular](url)

PedroChaparro commented:
> I agree with @PedroChaparro, I also feel that using a library is overkill, when a modal can be done with tailwind directly.

I think that you need to agree whether to use MaterialUI or style the page "manually" with Tailwind. It's just to keep the styles consistent across the page.

DanielEJaimes committed:
refactor: Modal Tailwind implementation

Modal changed from MUI implementation to Tailwind implementation
Uninstall of MUI libraries

PedroChaparro commented:
@MiguelMRojas Your review is pending here.

MiguelMRojas commented on src/components/UpdatePassword/UpdatePassword.tsx:
```diff
@@ -0,0 +1,111 @@
+                    />
+                    {/*Mensaje de contraseña insegura*/}
+                    {!isSecurePassword() ? (
+                      <span className="mr-1 inline-block rounded bg-red-200 px-1 py-0 text-xs font-semibold uppercase uppercase text-red-600 last:mr-0">
```
For the alerts we are using the following library:
`https://react-hook-form.com/`

DanielEJaimes replied:
Done in the commit "refactor: UpdatePassword"

MiguelMRojas commented on src/main.tsx:
```diff
@@ -6,6 +6,7 @@ import "./global.css";
 import { NavbarScreen } from "./components/Nav/Nav.tsx";
 import { Login } from "./screens/Login.tsx";
 import { Register } from "./screens/Register.tsx";
+// import { UpdatePassword } from "./components/UpdatePassword/UpdatePassword.tsx";
```
The **component** is called as such from the `screen` folder, not from the main file.

DanielEJaimes committed:
feat: Sidebar responsive

Sidebar component responsive

DanielEJaimes commented:
Sidebar component responsive

https://github.com/hawks-atlanta/frontend-react/assets/131629166/873ba9b1-e86c-42af-84c9-b0da642f9dd6

DanielEJaimes committed:
feat: File component

Creation of the file component with the dropdown menu to delete, edit, share, move

DanielEJaimes commented:
File component with dropdown menu

https://github.com/hawks-atlanta/frontend-react/assets/131629166/a344ff67-15b3-4fb4-96aa-7d4bc608ec01

MiguelMRojas commented:
Shouldn't the sidebar and the component file be in the same view?

> What do you think?

 @PedroChaparro

PedroChaparro commented:
> What do you think?

What do you mean?

DanielEJaimes committed:
feat: Homepage

Pagination still pending
Removed the sidebar component, added directly in the homepage screen
Design changes at File.tsx and UpdatePassword.tsx

DanielEJaimes commented:
Homepage

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/a57b6dbc-db85-4bbb-92aa-e4c67647a0df)

Mobile view

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/bfc10187-72d6-4f88-b4b0-6ee27caa3159)

Update password implemented

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/24a36f60-0487-451f-aada-60abbd2c2c70)

DanielEJaimes committed:
fix: 'NavbarScreen' is defined but never used

DanielEJaimes committed:
refactor: UpdatePassword

Implementation of update password with react hook form

DanielEJaimes commented:
Update password modal

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/2232d953-ce40-488f-8dcd-7f6885c07008)

Min length

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/945050cd-dfe4-41f7-ac45-f978351cf5fd)

Pattern

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/897f96bf-725e-4616-9698-a9d68234f042)

Password match

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/7f0e4069-b8ba-4d22-8ace-68e8ee157d1c)

DanielEJaimes committed:
fix: Package.json

Fixed conflicts in the package.json and package-lock.json
Re-design of updatepassword component

DanielEJaimes committed:
fix: package-lock.json

Fixed versioning problem

MiguelMRojas commented on src/components/Dropdown/Dropdown.tsx:
```diff
@@ -0,0 +1,72 @@
+        type="button"
+        onClick={() => setShowDropdown(!showDropdown)}
+      >
+        <svg
```
Remember that for the icons we are using the Lucide library

MiguelMRojas commented on src/components/File/File.tsx:
```diff
@@ -0,0 +1,26 @@
+      style={{ borderRadius: "8px" }}
+      onClick={handleClick}
+    >
+      <FileText style={{ width: "4rem", height: "4rem" }} />
```
Remember that we are using Tailwind

DanielEJaimes committed:
style:remove css and svg

Removed css styles and svg elements

DanielEJaimes committed:
fix: lint

Documents with prettier

DanielEJaimes committed:
feat: Pagination homepage

Pagination in the homepage

DanielEJaimes commented:
Pagination (Bottom right)
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/84f0c95c-7cd8-4ae0-ac63-72b21ecb7725)

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/85af6fb7-3b7f-4e78-b7e4-212c7b0c8d49)

MiguelMRojas commented:
### The main navbar does not come out, I think that adding the navbar component on each screen is a bad practice.

MiguelMRojas reviewed (changes requested):
https://github.com/hawks-atlanta/frontend-react/assets/115849391/336d2f68-63a6-4c7d-86e2-897d21ffccef

When you change page, the current navbar changes size, this should not happen, also, when you change page, the "Files" should be together, and not separated

DanielEJaimes committed:
fix: Type error homepage and pagination

Solved the type error problems in the homepage and the pagination component

Woynert reviewed (commented):
Hi there, good job so far!

I see this PR is about the **Password reset modal**, yet you have also added extra things: file listing, lateral bar, and pagination.

Try to keep future PRs consistent to it's original purpose so that it can be properly reviewed and merged into `dev`. For any new features you may create a new PR.

DanielEJaimes committed:
style: Navbar size

Correction of the navigation bar on the homepage that changes size when changing pages
Correction of the position of the "File" components

DanielEJaimes commented:
https://github.com/hawks-atlanta/frontend-react/assets/131629166/6d54cc7a-e90f-4cf1-b0d9-12ebf0825fae

PedroChaparro commented:
> Try to keep future PRs consistent to it's original purpose so that it can be properly reviewed and merged into `dev`. For any new features you may create a new PR.

💯. Additionally, please remember to maintain the current tests, updating them as needed, and add new tests when introducing new features. That will help to keep a sufficient coverage metric and prevent the features from breaking during development.

PedroChaparro committed:
Merge branch 'dev' into feat/home

PedroChaparro committed:
chore(deps): Update lock file

PedroChaparro committed:
chore(deps): Update lock file

PedroChaparro commented:
> [!IMPORTANT]  
> @DanielEJaimes, please note that since this branch was outdated, I've created some commits to update it and fix some `ci` errors. Therefore, consider stashing your local changes with `git stash` and updating your local branch with `git pull` before creating new commits to avoid potential git conflicts.

PedroChaparro commented:
>[!IMPORTANT]
> @shoriwe, what should we do with this pull request? Please consider the following:

1. It's not clear how many features are included in this single PR. 
2. There are two different navbars.
3. No new tests were added, and the previous tests are broken since Miguel's navbar was removed.
4. Styles are inconsistent. For instance, the login form uses a rounded button while the modal to update the password uses a square one.

shoriwe commented:
> Important @shoriwe, what should we do with this pull request? Please consider the following:
> 
> 1. It's not clear how many features are included in this single PR.
> 2. There are two different navbars.
> 3. No new tests were added, and the previous tests are broken since Miguel's navbar was removed.
> 4. Styles are inconsistent. For instance, the login form uses a rounded button while the modal to update the password uses a square one.

Why old navbar was removed?

shoriwe commented:
My only points will be create a new branch based on the last commit were this PR was working properly, then merge that branch to `dev` finally to `main`

PedroChaparro commented:
> Why old navbar was removed?


It seems that they're trying to use a different navbar for the user icon instead of updating the navbar state according to the user's session. 

|![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/ceb3430e-b58c-4524-808a-1c386d0d10f8)| ![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/bcaa1a33-47e9-48f2-8329-7efcbe692336)|
|--------|--------|

Even the height and colors are different.

PedroChaparro commented:
I'm closing this pull request due to the reasons mentioned above.

@DanielEJaimes, please follow @shoriwe's advice. I recommend that you start by creating a new PR specifically for the files page, including the sidebar and file cards, but excluding the navbar (#31). After that, you both can decide what to do with the navbar and modal in separated PRs.
